### ISSUE TITLE
Interested in generating a high-order mesh using methods to be used with mfem::Mesh init constructor

### ISSUE THREAD

**Issue by junkudo** [mfem/mfem#996]
Hey,
I'm interested in be able to generate high-order curved meshes using the exposed API.  I've previously been able to generate linear meshes using the methods intended to be used the init constructor (i.e. `AddTri (...)`, `AddTet(...)`) but I recognize that these methods are only constructing the linear geometries without any notion of high-order coordinates.  From a cursory look at the mesh implementation, it looks like I'll need to somehow create and initialize `mfem::Mesh::Nodes`.  Is what I'm trying to do currently possible?  If not, do you have any suggestions on how to make high-order meshes with code only?

CC @Dan2997925

**Comment by tzanio** [comment]
Hi @junkudo,

One way to define high-order meshes is to define a low-order mesh, increase the curvature, and map it with `Mesh::Transform`, see e.g. the Toroid miniapp:

https://github.com/mfem/mfem/blob/master/miniapps/meshing/toroid.cpp#L146-L151

This only works for relatively simple geometries. If you need a full high-order mesh generator, see the PUMI integration (cc: @MarkShephard) as well as the GMsh and Cubit support.

Hope this helps,
Tzanio

**Comment by MarkShephard** [comment]
To @junkudo,

If your long term plans are to regularly work with complex geometries for which you want curved meshes, you may want to look into a workflow we have developed that can go from CAD to MFEM. I would be happy to have a conference call to get a better understanding of what you want to be able to do in the long run and to give an introduction to some of the options that could be considered.

**Comment by junkudo** [comment]
Hi Tzanio,
What if I know where the high-order points should be?  As in, I have the entire mesh known during run time and I want to create a mfem::Mesh object that represents that mesh.

**Comment by tzanio** [comment]
Do you need a copy-constructor for `Mesh`? In principle the nodes are stored in the `Nodes` grid function inside the mesh, so you can just copy that.

**Comment by junkudo** [comment]
Sorry for dancing around the specifics.  I have an existing curved mesh and I want to apply a defined linear transformation to it resulting in a new mesh.  The way I set out to do this was to leave the original mesh alone and create a new mesh with the new transformed coordinates (but same connectivity).  This works fine for the linear meshes because I can directly create linear meshes with the transformed coordinates.  I'm not sure how to accomplish this for curved meshes.  Based on the transformation, I know where the high-order points should be located but I don't know how to specify them through the creation API.  Can I directly modify the high-order nodal points?  If I could do that, I can copy the mesh and then modify those points myself.

@MarkShephard thanks for the offer but I actually think my application is much simpler than actually generating curved meshes.   Also, it's a small world.  I've worked with PUMI before at my previous job... 😄
